src/App.tsx

```diff
@@ -1,93 +1,104 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+import { TodosContext } from './TodosContext';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { useLocalStorage } from './hooks/useLocalStorage';
 
 export const App: React.FC = () => {
-  return (
-    <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+
+  const [status, setStatus] = useState(Status.all);
+  const [isAllChecked, setIsAllChecked] = useState(false);
+  const [value, setValue] = useState('');
 
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
+  const isAllCompleted = todos.every((todo: Todo) => todo.completed);
 
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = event.target.value.trim();
 
-        <ul className="todo-list" data-cy="todoList">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    setValue(newValue === '' ? '' : event.target.value);
+  };
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-completed" />
-              <label htmlFor="toggle-completed">qwertyuio</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const handleInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (value.length && event.key === 'Enter') {
```

Comment:
```suggestion
    if (value && event.key === 'Enter') {
```
if value is not empty string it will be true boolean value